src/View/Form/EntityContext.php

```diff
@@ -259,7 +260,7 @@ public function val($field, $options = [])
 
             return $this->_schemaDefault($part, $entity);
         }
-        if (is_array($entity)) {
+        if (is_array($entity) || $entity instanceof ArrayAccess) {
             $key = array_pop($parts);
```

Comment:
Please note that array_pop will not work with ArrayAccess
https://stackoverflow.com/questions/8788002/arrayaccess-and-native-php-array-functions